nuxeo-runtime/nuxeo-runtime-stream/src/main/java/org/nuxeo/runtime/stream/StreamProcessorDescriptor.java

```diff
@@ -55,6 +56,41 @@ public String getId() {
         }
     }
 
+    @XObject(value = "filter")
+    public static class FilterDescriptor implements Descriptor {
+
+        @XNode("@name")
+        public String name;
+
+        @Override
+        public String getId() {
+            return name;
+        }
+
+        // To provide a custom retry policy
+        @XNode("@class")
+        public Class<? extends RecordFilter> klass;
+
+        @XNodeMap(value = "option", key = "@name", type = HashMap.class, componentType = String.class)
+        public Map<String, String> options = new HashMap<>();
+
+        public RecordFilter getFilter() {
+            if (!RecordFilter.class.isAssignableFrom(klass)) {
+                throw new IllegalArgumentException("Cannot create filter: " + getId() + " for stream: " + this.getId()
+                        + ", class must implement Filter");
+            }
+            try {
+                RecordFilter ret = klass.getDeclaredConstructor().newInstance();
+                ret.init(options);
```

Comment:
What do you think to move the init/options to the constructor?

Reply:
as discussed it can stay like this to force configuration from a map and still enable to create them with custom constructor (at least at the nuxeo stream lib level)